dateutil/parser.py

```diff
@@ -602,8 +602,6 @@ def parse(self, timestr, default=None, ignoretz=False, tzinfos=None, **kwargs):
                     raise ValueError("Offset must be tzinfo subclass, "
                                      "tz string, or int offset.")
                 ret = ret.replace(tzinfo=tzinfo)
-            elif res.tzname and res.tzname in time.tzname:
```

Comment:
This would break compatibility, I'm pretty sure we shouldn't do this as implemented. The most I'd be comfortable with is moving this down the list as the last resort rather than one of the first.